.github/workflows/ci.yml

```diff
@@ -65,8 +65,9 @@ jobs:
     name: Build Babel Artifacts
     needs: prepare-yarn-cache
     runs-on: ubuntu-latest
-    env:
-      YARN_NODE_LINKER: pnp # use pnp linker for better linking performance and stricter checks
+    # Yarn PnP does not support native ESM yet (https://github.com/yarnpkg/berry/issues/638)
+    # env:
+    #    YARN_NODE_LINKER: pnp # use pnp linker for better linking performance and stricter checks
```

Comment:
PnP doesn't support ESM yet, but they are working on it: https://github.com/yarnpkg/berry/pull/2161

I personally wouldn't expect them to introduce support for ESM by default until Node.js' loader API becomes stable (currently it's `--experimental-loader`), but:
- We could decide to use https://github.com/yarnpkg/berry/pull/2161 instead of their last release: Yarn is committed into our repo, so it will not "suddently break" anyway.
- They could release ESM support even if loaders are still experimental, maybe behind an `experimentalESM` config option (cc @merceyz, just in case you like this suggestion :stuck_out_tongue:)
- They already use our repository in their E2E tests for `node_modules`: also giving them a real-world repository that use Yarn 2 to test their ESM support _might_ help with getting ESM support sooner and/or make it more stable!

Reply:
All that PR needs is someone to test it, would be great if a large and real project like Babel could do that

Reply:
I'll try using it this week and report feedback!

Reply:
It enables itself when the root `package.json` has `type: module` but I can turn that off

Reply:
Any updates on ESM support in Yarn 2? IMO the PnP check is important in case we forget declaring dependencies. We could add a new build job which transpiles Gulpfile.mjs and then build with PnP.

Reply:
Would https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-extraneous-dependencies.md be enough? It also has the advantage that catches the error locally other than on CI.

Reply:
Actually, we already use it: https://github.com/babel/babel/blob/9ac36b136adcfa1940cdcc60399cb4079fb67361/.eslintrc.js#L44

Reply:
Then I am wondering why it did not catch https://github.com/babel/babel/pull/12011/files#r546151754.

Reply:
Because we only lint `src/` files :grimacing:

Reply:
It's less important to lint other things since they will never affect our users, but we can lint everything if you prefer.

Reply:
👍 to lint everything except git-ignored.

Reply:
> Any updates on ESM support in Yarn 2?

@JLHwung It works as far as I can tell, @nicolo-ribaudo was testing it on the Babel repo but I don't know the result of that test. All the issues that were brought up was fixed as far as I can tell

Reply:
Yup, in this repo it's working (or at least, it was a few weeks ago).
However, I didn't want to put any pressure on yarn since the ESM loader API is far from being stable.